frontend/src/Components/ContentComponent.tsx

```diff
@@ -59,7 +60,34 @@ function updateImg(img: HTMLImageElement) {
             img.parentElement?.replaceWith(iframe);
         });
     }
-    if (ytUrl) {
+    return !!ytUrl;
+}
+
+/**
+ * Convert mp4 video embeds into video elements
+ */
+function processVideoEmbed(img: HTMLImageElement) {
+    const videoUrl = img.dataset.video;
+
+    if (videoUrl && !img.classList.contains('video-embed-processed')) {
+        img.classList.add('video-embed-processed');
+        img.addEventListener('click', (e) => {
+            e.preventDefault();
+            const video = document.createElement('video');
+            video.src = videoUrl;
+            video.controls = true;
+            video.autoplay = true;
+            video.loop = !!img.dataset.loop;
+            video.style.width = img.width.toString() + 'px';
```

Comment:
Won't work if poster is absent or generic placeholder was used

Reply:
This logic expects `placeholder` from TheParser.